R/Logging.R

```diff
@@ -84,10 +84,16 @@ createConsoleAppender <- function(layout = layoutSimple) {
 #' @export
 createFileAppender <- function(layout = layoutParallel, fileName) {
   appendFunction <- function(this, level, message) {
-    con <- file(fileName, open = "at", blocking = FALSE)
-    writeLines(text = message, con = con)
-    flush(con)
-    close(con)
+    tryCatch({
+      con <- file(fileName, open = "at", blocking = FALSE)
+      writeLines(text = message, con = con)
+      flush(con)
+      close(con)
+    },
+    error = function(e) {
+      closeAllConnections()
```

Comment:
Just curious: why close ALL connections? The program might have a few connections open. Wouldn't this interfere with that?

Reply:
Your right, I did not thought about that since my program only uses ODBC and no file connections. Maybe there is a way to only close the broken connections that where created due to the file opening, or check if the file is accessible before writing.

But without this check the logging causes the whole program to shut down if the log folder gets delete while running the program.

Reply:
I think [this](https://github.com/OHDSI/ParallelLogger/commit/a31a89826906e4bebb849936c685868811576a31) is what you really want: if the log folder ceases to exist, a warning is thrown once to alert the user of this fact, and the file appender is deleted to avoid future warnings. There's no need to close any connections , because the connection is already closed after every write operation.

Could you try this out? You can install using `devtools::install_github("ohdsi/ParallelLogger")`.

Reply:
I've tried it (only one small test, busy day, sorry) and it seams like it works exactly the way i want it to: The file appender gets deleted and the consol appender still works. 

Thank you very much.
Will these changes also be transferred to CRAN? How long dose it usually take?